Lib/idlelib/idle_test/test_autocomplete.py

```diff
@@ -132,12 +132,16 @@ def test_fetch_completions(self):
         # a small list containing non-private variables.
         # For file completion, a large list containing all files in the path,
         # and a small list containing files that do not start with '.'
-        pass
+        small, large = self.autocomplete.fetch_completions(
+                '', ac.COMPLETE_ATTRIBUTES)
+        self.assertLess(len(small), len(large))
+        self.assertTrue('AutoComplete' not in small or  # See issue 36405.
```

Comment:
I dislike complex expressions in assertTrue() because it gives little information when the test is failed and it is easy to make an error that makes the test be always passed.

If I understand the problem correctly, this could be written as:
```python
if __main__ is idlelib.autocomplete:
    self.assertIn('testing_in_autocomplete', small)
else:
    self.assertNotIn('AutoComplete', small)
```

Reply:
I agree about disliking what is a bit of a kludge.  However, even when autocomplete.py is run, __main__ and idlelib.autocomplete (ac) are different and unequal module instances, with different __names__ ('__main__' and 'idlelib.autocomplete'), different types for __builtins__ (module and dict), and unique attributes (__annotations__ and __cached__).  What are equal if and only if both modules derived from the same file are the __file__s. Test code added to test_autocomplete.
```python
        print(__main__, ac, __main__ is ac)
        print(type(__main__.__builtins__), type(ac.__builtins__))
        print(dir(__main__))
        print(dir(ac))
        print(__main__.__file__, ac.__file__)
```
 The goal is to detect whether the merged dict is __main__.__dict__ (right) or autocomplete globals() (wrong).  When the test is run with autocomplete as main, they are the same, so there is nothing to test.  We must either skip `'AutoComplete not in small'` or add something to make the resulting expression true.  I did the latter, but former will be nicer.  The following works.
```python
    if __main__.__file__ != ac.__file__:
        self.assertNotIn('AutoComplete', small)
```
Thank you for the hint of where to look.